src/api/wallet/WalletApiImpl.ts

```diff
@@ -248,13 +247,14 @@ export class WalletApiImpl implements WalletApi {
       isConnected,
       userAddress: accounts[0],
       networkId: isConnected ? +chainId : undefined,
+      web3: this._web3,
     }
   }
 
   private async _notifyListeners(): Promise<void> {
-    await Promise.resolve()
+    const blockNumber = await this._blockNumber
     const walletInfo: WalletInfo = this._getWalletInfo()
-    this._listeners.forEach(listener => listener(walletInfo))
+    this._listeners.forEach(listener => listener({ ...walletInfo, blockNumber }))
```

Comment:
```js
{ ...walletInfo, blockNumber }
```
will get recreated for each listener, construct outside the loop

Reply:
:+1: